### PR TITLE
allow disable shadow map auto update, fix shadow system init

### PR DESCRIPTION
**Pull request by ngokevin**
**Description:**

**Changes proposed:**
- Shadow system setup was never getting called.
- Allow disable auto update shadow map to allow one-time shadow baking and on-demand updates.

### CODE REVIEW

File: docs/components/shadow.md [hunk 1]
```diff
@@ -37,3 +36,16 @@ component for more information.
...
+
+## Scene Properties
+
+The shadow system exposes scene-level properties for configuring the renderer
+for shadows. These are set on `<a-scene>` (e.g., `<a-scene shadow="autoUpdate:
+  false">`).
+
+| Property           | Description                                                                                                                                                                                           | Default Value                       |
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------|
+| autoUpdate         | Whether to dynamically update the shadow map every frame. Disable and manually update by setting `renderer.shadowMap.needsUpdate = true` for best performance. Calculating shadow maps are expensive. | true                                |
```
Comment (donmccurdy) [inline docs/components/shadow.md]: nit: s/are expensive/is expensive

**Comment by dmarcos** [comment]
Tests failing

**Comment by ngokevin** [comment]
Updated.

`needs test fixed` label? GitHub shows red X everywhere already.

**Comment by dmarcos** [comment]
It helps me visually and sometimes you get an `X` on the pull request list when in reality is not broken tests but some sort of code coverage compliance thing.

**Comment by ngokevin** [comment]
OK I don't like more process, but it's fine. FYI those code coverage compliance things were from when people accidentally included `.only` in their tests (46c7cfcd12f6ef4bf3d30f6c744035ffd173510f). I've written a hook to prevent that.

**Pull request merged**